File: javascript/leetcode/832. Flipping an Image.js
/**
 * @param {number[][]} A
 * @return {number[][]}
 */
var flipAndInvertImage = function (A) {
  A.forEach((el, ind) => {
    let start = 0;
    let end = el.length - 1;

    while (start <= end) {
      A[ind][start] = A[ind][start] === 1 ? 0 : 1;
      if (start === end) break;

      A[ind][end] = A[ind][end] === 1 ? 0 : 1;

      const temp = A[ind][start];
      A[ind][start++] = A[ind][end];
      A[ind][end--] = temp;
    }
  });

  return A;
};

File: c++/leetcode/832. Flipping an Image.cpp
class Solution
{
public:
  vector<vector<int>> flipAndInvertImage(vector<vector<int>> &A)
  {
    for (int i = 0; i < A.size(); i++)
    {
      int start = 0;
      int end = A[i].size() - 1;

      while (start <= end)
      {
        A[i][start] = A[i][start] == 1 ? 0 : 1;
        if (start == end)
          break;

        A[i][end] = A[i][end] == 1 ? 0 : 1;

        // swap
        int temp = A[i][start];
        A[i][start++] = A[i][end];
        A[i][end--] = temp;
      }
    }

    return A;
  }
};